src/main/java/org/spongepowered/common/mixin/core/world/gen/MixinChunkProviderServer.java

```diff
@@ -116,6 +116,15 @@ public ChunkDataStream getGeneratedChunks() {
         return WorldStorageUtil.doesChunkExist(this.world, this.chunkLoader, chunkCoords);
     }
 
+    // This method exists to allow us to speed up world pre generation.
+    @Override
+    public boolean doesChunkExistSync(Vector3i location) {
```

Comment:
[It's called here, as part of a `Predicate<Vector31>`](https://github.com/SpongePowered/SpongeCommon/pull/1149/files#diff-7db96f1d7969f596d94f9dfd4f2a99f2R118) - it's then called as part of the [`areAllChunksLoaded` routine](https://github.com/SpongePowered/SpongeCommon/pull/1149/files#diff-7db96f1d7969f596d94f9dfd4f2a99f2R297). I've done it like this to avoid repeated `instanceof` checks when we only really need one.